symbolic_pymc/tensorflow/printing.py

```diff
@@ -12,23 +12,36 @@
 class TFlowPrinter(object):
     """A printer that indents and keeps track of already printed subgraphs."""
 
-    def __init__(self, formatter, buffer):
+    def __init__(self, formatter, buffer, depth_index):
         self.buffer = buffer
         self.formatter = formatter
+        self.depth_count = 0
         self.indentation = ""
         self.printed_subgraphs = set()
+        try:
+            self.depth_index = (
+                range(depth_index[0], depth_index[1]) if None not in depth_index else None
+            )
+        except TypeError:
+            raise ValueError("depth_index needs to be a tuple or list of integers")
 
     @contextmanager
     def indented(self, indent):
         pre_indentation = self.indentation
+        self.depth_count += 1
         if isinstance(indent, int):
             self.indentation += " " * indent
         else:
             self.indentation += indent
+
+        if self.depth_index and "... " not in self.indentation:
```

Comment:
Why do you need to search and replace in the strings you're responsible for creating?!  Why can't you explicitly craft the correct value of `self.indentation` in the first place?

Reply:
I removed the search in the latest commit.